Teams with external security groups 'panics' on apply

**Describe the bug**
I cannot apply changes to an Octopus Team when changing or adding an external security group The plan works fine, and shows me the correct change that will be made, however the apply fails

**Steps to reproduce**
1. Create a Team and specify a valid external security group in Terraform
2. Perform standard terraform init/plan to confirm changes
3. Execute terraform apply
4. See error below :)

**Expected behavior**
Adding the group manually in the UI works, it does not work via terraform 

**Logs and other supporting information**
root@EW1B-P-TF02:/tmp/TF/Octopus Deploy/0.0.11/AWS-PRO# terraform15 apply -var-file=variables-pro.tfvars -target=octopusdeploy_team.devops_permissions_reporter
octopusdeploy_team.devops_permissions_reporter: Refreshing state... [id=Teams-381]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following
symbols:
  ~ update in-place

Terraform will perform the following actions:

  # octopusdeploy_team.devops_permissions_reporter will be updated in-place
  ~ resource "octopusdeploy_team" "devops_permissions_reporter" {
        id                 = "Teams-381"
        name               = "Permissions Reporter"
        # (5 unchanged attributes hidden)

      + external_security_groups {
          + display_name = "BERMUDA\\DevOps"
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.
╷
│ Warning: Resource targeting is in effect
│
│ You are creating a plan with the -target option, which means that the result of this plan may not represent all of the changes
│ requested by the current configuration.
│
│ The -target option is not for routine use, and is provided only for exceptional situations such as recovering from errors or
│ mistakes, or when Terraform specifically suggests to use it as part of an error message.
╵

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

octopusdeploy_team.devops_permissions_reporter: Modifying... [id=Teams-381]
╷
│ Warning: Applied changes may be incomplete
│
│ The plan was created with the -target option in effect, so some changes requested in the configuration may have been ignored and
│ the output values may not be fully updated. Run the following command to verify that no other changes are pending:
│     terraform plan
│
│ Note that the -target option is not suitable for routine use, and is provided only for exceptional situations such as recovering
│ from errors or mistakes, or when Terraform specifically suggests to use it as part of an error message.
╵
╷
│ Error: Plugin did not respond
│
│   with octopusdeploy_team.devops_permissions_reporter,
│   on devops_teams.tf line 72, in resource "octopusdeploy_team" "devops_permissions_reporter":
│   72: resource "octopusdeploy_team" "devops_permissions_reporter" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may
│ contain more details.
╵

Stack trace from the terraform-provider-octopusdeploy_v0.7.36 plugin:

panic: interface conversion: interface {} is []interface {}, not *schema.Set

goroutine 40 [running]:
github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy.expandTeam(0xc0008f7700, 0x19)
        github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy/schema_team.go:38 +0x5e9
github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy.resourceTeamUpdate(0xfb5ac0, 0xc00040df20, 0xc0008f7700, 0xd21100, 0xc0000a2a00, 0xc0005a88e0, 0x93c4ea, 0xc0005f2b80)
        github.com/OctopusDeploy/terraform-provider-octopusdeploy/octopusdeploy/resource_team.go:85 +0xf6
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xc00055ac40, 0xfb5a40, 0xc000422a80, 0xc0008f7700, 0xd21100, 0xc0000a2a00, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.0/helper/schema/resource.go:364 +0x17c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00055ac40, 0xfb5a40, 0xc000422a80, 0xc0002e7880, 0xc0005f2b80, 0xd21100, 0xc0000a2a00, 0x0, 0x0, 0x0, ...)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.0/helper/schema/resource.go:464 +0x390
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0002ea6e0, 0xfb5a40, 0xc000422a80, 0xc0007ae2d0, 0xc000422a80, 0x0, 0xfc1580)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.6.0/helper/schema/grpc_provider.go:955 +0x8cf
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc0003fd360, 0xfb5a40, 0xc000422a80, 0xc0002e76c0, 0xc0003fd360, 0xc000477bc0, 0xc00057bba0)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0xae
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0xe4b860, 0xc0003fd360, 0xfb5b00, 0xc000477bc0, 0xc00040d440, 0x0, 0xfb5b00, 0xc000477bc0, 0xc0008702c0, 0x2ac)
        github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00032a8c0, 0xfbf080, 0xc0002c0600, 0xc000471e60, 0xc00051cde0, 0x14c7bc0, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.37.0/server.go:1217 +0x522
google.golang.org/grpc.(*Server).handleStream(0xc00032a8c0, 0xfbf080, 0xc0002c0600, 0xc000471e60, 0x0)
        google.golang.org/grpc@v1.37.0/server.go:1540 +0xd05
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000292180, 0xc00032a8c0, 0xfbf080, 0xc0002c0600, 0xc000471e60)
        google.golang.org/grpc@v1.37.0/server.go:878 +0xa5
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.37.0/server.go:876 +0x1fd

Error: The terraform-provider-octopusdeploy_v0.7.36 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.

**Screenshots**
n/a

**Environment and versions:**
- OS: Linux_amd64
- Octopus Server Version: 2020.4.11
- Terraform Version: Terraform v0.15.1
- Octopus Terraform Provider Version: v0.7.36

**Additional context**

My definition is as follows:

resource "octopusdeploy_team" "devops_permissions_reporter" {
  name = "Permissions Reporter"
  external_security_groups {
    display_name = "DevOps"
  }
}
}

--- a/docs/data-sources/teams.md
+++ b/docs/data-sources/teams.md
@@ -39,14 +39,14 @@ Read-Only:
 - **can_change_members** (Boolean)
 - **can_change_roles** (Boolean)
 - **description** (String) The user-friendly description of this team.
-- **external_security_groups** (List of Object) (see [below for nested schema](#nestedatt--spaces--external_security_groups))
+- **external_security_group** (List of Object) (see [below for nested schema](#nestedatt--spaces--external_security_group))
 - **id** (String) The unique ID for this resource.
 - **name** (String) The name of this team.
 - **space_id** (String) The space associated with this team.
 - **users** (List of String) A list of user IDs designated to be members of this team.
 
-<a id="nestedatt--spaces--external_security_groups"></a>
-### Nested Schema for `spaces.external_security_groups`
+<a id="nestedatt--spaces--external_security_group"></a>
+### Nested Schema for `spaces.external_security_group`
 
 Read-Only:
 
@@ -66,14 +66,14 @@ Read-Only:
 - **can_change_members** (Boolean)
 - **can_change_roles** (Boolean)
 - **description** (String) The user-friendly description of this team.
-- **external_security_groups** (List of Object) (see [below for nested schema](#nestedatt--teams--external_security_groups))
+- **external_security_group** (List of Object) (see [below for nested schema](#nestedatt--teams--external_security_group))
 - **id** (String) The unique ID for this resource.
 - **name** (String) The name of this team.
 - **space_id** (String) The space associated with this team.
 - **users** (List of String) A list of user IDs designated to be members of this team.
 
-<a id="nestedatt--teams--external_security_groups"></a>
-### Nested Schema for `teams.external_security_groups`
+<a id="nestedatt--teams--external_security_group"></a>
+### Nested Schema for `teams.external_security_group`
 
 Read-Only:
 

--- a/docs/resources/team.md
+++ b/docs/resources/team.md
@@ -26,13 +26,13 @@ This resource manages teams in Octopus Deploy.
 - **can_change_members** (Boolean)
 - **can_change_roles** (Boolean)
 - **description** (String) The user-friendly description of this team.
-- **external_security_groups** (Block List) (see [below for nested schema](#nestedblock--external_security_groups))
+- **external_security_group** (Block List) (see [below for nested schema](#nestedblock--external_security_group))
 - **id** (String) The unique ID for this resource.
 - **space_id** (String) The space associated with this team.
 - **users** (List of String) A list of user IDs designated to be members of this team.
 
-<a id="nestedblock--external_security_groups"></a>
-### Nested Schema for `external_security_groups`
+<a id="nestedblock--external_security_group"></a>
+### Nested Schema for `external_security_group`
 
 Optional:
 

--- a/octopusdeploy/schema_team.go
+++ b/octopusdeploy/schema_team.go
@@ -34,8 +34,8 @@ func expandTeam(d *schema.ResourceData) *octopusdeploy.Team {
 		team.Description = v.(string)
 	}
 
-	if v, ok := d.GetOk("external_security_groups"); ok {
-		team.ExternalSecurityGroups = expandExternalSecurityGroups(v.(*schema.Set).List())
+	if v, ok := d.GetOk("external_security_group"); ok {
+		team.ExternalSecurityGroups = expandExternalSecurityGroups(v.([]interface{}))
 	}
 
 	if v, ok := d.GetOk("space_id"); ok {
@@ -55,16 +55,16 @@ func flattenTeam(team *octopusdeploy.Team) map[string]interface{} {
 	}
 
 	return map[string]interface{}{
-		"can_be_deleted":           team.CanBeDeleted,
-		"can_be_renamed":           team.CanBeRenamed,
-		"can_change_members":       team.CanChangeMembers,
-		"can_change_roles":         team.CanChangeRoles,
-		"description":              team.Description,
-		"external_security_groups": flattenExternalSecurityGroups(team.ExternalSecurityGroups),
-		"id":                       team.GetID(),
-		"name":                     team.Name,
-		"space_id":                 team.SpaceID,
-		"users":                    team.MemberUserIDs,
+		"can_be_deleted":          team.CanBeDeleted,
+		"can_be_renamed":          team.CanBeRenamed,
+		"can_change_members":      team.CanChangeMembers,
+		"can_change_roles":        team.CanChangeRoles,
+		"description":             team.Description,
+		"external_security_group": flattenExternalSecurityGroups(team.ExternalSecurityGroups),
+		"id":                      team.GetID(),
+		"name":                    team.Name,
+		"space_id":                team.SpaceID,
+		"users":                   team.MemberUserIDs,
 	}
 }
 
@@ -122,7 +122,7 @@ func getTeamSchema() map[string]*schema.Schema {
 			Optional:    true,
 			Type:        schema.TypeString,
 		},
-		"external_security_groups": {
+		"external_security_group": {
 			Optional: true,
 			Elem:     &schema.Resource{Schema: getExternalSecurityGroupsSchema()},
 			Type:     schema.TypeList,
@@ -156,8 +156,8 @@ func setTeam(ctx context.Context, d *schema.ResourceData, team *octopusdeploy.Te
 	d.Set("can_change_roles", team.CanChangeRoles)
 	d.Set("description", team.Description)
 
-	if err := d.Set("external_security_groups", flattenExternalSecurityGroups(team.ExternalSecurityGroups)); err != nil {
-		return fmt.Errorf("error setting external_security_groups: %s", err)
+	if err := d.Set("external_security_group", flattenExternalSecurityGroups(team.ExternalSecurityGroups)); err != nil {
+		return fmt.Errorf("error setting external_security_group: %s", err)
 	}
 
 	d.Set("id", team.GetID())

